### ISSUE TITLE
SelectBooleanCheckbox, ToggleSwitch: NVDA (Screen reader) reads "empty" when focusing

### ISSUE THREAD

**Issue by gipoezcan** [primefaces/primefaces#12577]
### Describe the bug

The NVDA screen reader doesn't recognize the labels for the checkbox and the toggle switch

### Reproducer

1. Activate NVDA
2. go to [the checkbox demo page](http://www.primefaces.org/showcase/ui/input/booleanCheckbox.xhtml) or the [toggle switch demo page](http://www.primefaces.org/showcase/ui/input/toggleSwitch.xhtml?jfwid=a29d8)
3. when you focus on the checkbox or the toggle switch, NVDA just announces "empty" ("leer" in German).

### Expected behavior

The labels of both will be announced

### PrimeFaces edition

Community

### PrimeFaces version

14

### JSF implementation

Mojarra

### JSF version

any

### Java version

any

### Browser(s)

Firefox and Chrome (any version) under Windows 10 and NVDA 2024_3

### Our Workaround

    html .ui-helper-hidden-accessible {
      border: unset;
      clip: unset;
      height: unset;
      margin: unset;
      overflow: unset;
      padding: unset;
      width: unset;
      pointer-events: none;
      opacity: 0;
    }
    
    .ui-helper-hidden-accessible input {
      transform: unset !important;
    }

### Related

https://github.com/primefaces/primefaces/issues/6561

**Comment by melloware** [comment]
nice @gipoezcan its actually related to these two: https://github.com/primefaces/primefaces/pull/11639

We have been blaming NVDA:  https://github.com/nvaccess/nvda/issues/16300

**Comment by melloware** [comment]
so what is interesting about you doing this:

```css
border: unset;
  clip: unset;
  height: unset;
  margin: unset;
  overflow: unset;
  padding: unset;
  width: unset;
```

That is not what any other CSS library from Bootstrap, to Jquery, to Tailwind do.  They all do this.

```css
/* https://stackoverflow.com/a/62109988/502366 */
.ui-helper-hidden-accessible {
    border: 0 !important;
    padding: 0 !important;
    margin: -1px !important;
    position: absolute !important;
    height: 1px !important;
    width: 1px !important;
    overflow: hidden !important;
    clip: rect(1px, 1px, 1px, 1px) !important; /* Legacy Browsers */
    clip-path: inset(50%) !important; /* Modern Browsers */
    white-space: nowrap !important;
}
```

**Comment by melloware** [comment]
so it sounds like NVDA just wants this.

```css
.ui-helper-hidden-accessible {
   pointer-events: none;
  opacity: 0;
}
```

**Comment by melloware** [comment]
cc @f1nnk  @Weavelor see above

**Comment by gipoezcan** [comment]
@melloware Thank you for the research! To be honest, I tried everything possible until it seemed to work, first unsetting anything coming from Primefaces to start from scratch (apart from position). Other screen readers don't seem to have any problems with this but we needed something that also worked with NVDA, because users started reporting problems already and we couldn't wait for an NVDA release. I would understand if Primefaces waits for NVDA potentially fixing this but NVDA has 2.2K open bugs so I didn't open a ticket there, maybe I should have...

**Comment by melloware** [comment]
If you have tested with other screen readers besides NVDa then I really like this change.

**Comment by gipoezcan** [comment]
Jaws was already working and seems to keep working after my workaround. NVDA only works with the workaround. I didn't test with anything else. Is there a list of screen readers Primefaces developers test with?

**Comment by melloware** [comment]
OK PR submitted.

**Comment by f1nnk** [comment]
@melloware Have you tried this with the Advanced SelectOneListbox in Firefox? The problem still persists for me with the fix.
What i didn't really mention in my original issue is that the selecting seems to work differently in Firefox than Chrome for example. When you tab into it once, the selection works normally but NVDA doesn't read out the elements when you go through it with the arrow keys. When you tab into it twice, the selection doesn't work visually but NVDA reads out the elements.

**Comment by melloware** [comment]
also @f1nnk that SelectOneListBox doesn't use the `ui-hidden` so it has nothing to do with this fix.  The Advanced example uses ARIA tags to give the browser hints on the reading.

**Comment by f1nnk** [comment]
@melloware ah okay, just wanted to say that #11623 is not fixed since you mentioned it in your [PR](https://github.com/primefaces/primefaces/pull/12583)

**Comment by melloware** [comment]
Thanks re-opened that one and asked some follow up questions.
